### PR TITLE
adds canonical to meta()

### PR DESCRIPTION
**Pull request by thinkingmedia**
Adds support for `<link rel="canonical" href="...">` for `Html->meta()`

This is a Google standard tag for indicating duplicate content.

### CODE REVIEW

**Comment by saeideng** [comment]
master or 3.next branch?

**Comment by thinkingmedia** [comment]
this can go to master. should be non-breaking.

**Comment by markstory** [comment]
Looks good. Needs a test though 💍

**Comment by thinkingmedia** [comment]
@markstory I'll add "prev" and "next" as well plus tests tomorrow.

File: tests/TestCase/View/Helper/HtmlHelperTest.php [hunk 1]
```diff
@@ -1637,6 +1640,35 @@ public function testMeta()
...
+     *
+     * @param string $type
+     * @param array $url
+     * @param string $expected_url
+     * @dataProvider dataMetaLinksProvider
+     */
+    public function testMetaLinks($type, array $url, $expected_url)
+    {
+        $result = $this->Html->meta($type, $url);
+        $expected = ['link' => ['href' => $expected_url, 'rel' => $type]];
```
Comment (thinkingmedia) [inline tests/TestCase/View/Helper/HtmlHelperTest.php]: One day I'm going to meet up with stickler in a dark alley.

Comment (ADmad) [inline tests/TestCase/View/Helper/HtmlHelperTest.php]: Or you could use codesniffer locally to prevent you from going down the dark path :)

Comment (inoas) [inline tests/TestCase/View/Helper/HtmlHelperTest.php]: `pre-commit` or `pre-push` @ https://git-scm.com/book/it/v2/Customizing-Git-Git-Hooks#Client-Side-Hooks ?

**Pull request merged**